### PR TITLE
Rewrite IAM Integration

### PR DESCRIPTION
**Pull request by ianstanton**
- Add clear steps for integrating IAM roles with Airflow pods on EKS
- Update `v0.15` and `v0.16` docs
- Related to https://github.com/astronomer/issues/issues/1834

### CODE REVIEW

**Comment by ianstanton** [comment]
Looks like there are some conflicts I'll need to resolve.

File: enterprise/v0.16/05_customize-airflow/06_integrate-iam.md [hunk 1]
```diff
@@ -6,151 +6,234 @@ description: "How to append IAM roles to an Airflow Deployment on Astronomer."
...
+On Astronomer, IAM roles can be appended to the Webserver, Scheduler and Worker pods within any individual Airflow Deployment on the platform.
 
-IAM roles on [AWS](https://aws.amazon.com/iam/faqs/), [GCP](https://cloud.google.com/iam/docs/overview) and other platforms are often used to manage the level of access a specific user (or object, or group of users) have to some resource (or set of resources). The resource in question could be an S3 bucket or Secret Backend, both of which are commonly used in tandem with Airflow and Astronomer and can now be configured to be accessible only to a subset of Kubernetes pods within your wider Astronomer cluster.
-
-To support this functionality, each pod in an Astronomer namespace (each of which maps to an Airflow deployment) is launched by a [Kubernetes Service Account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/). By annotating the service account with the role, all pods launched will inherit that role.
-
-At a high-level, you can:
-
-* Create an IAM role defining access to the target service or set of services (e.g. AWS S3)
```
Comment (paolaperaza) [inline enterprise/v0.16/05_customize-airflow/06_integrate-iam.md:17]: @ianstanton Think it's worth replacing this intro with something else or nah? Part of me thinks this high-level context for how you can get this to work on the platform is helpful but I'm also happy to defer to you if you feel like it's not

File: enterprise/v0.16/05_customize-airflow/06_integrate-iam.md [hunk 2]
```diff
@@ -6,151 +6,234 @@ description: "How to append IAM roles to an Airflow Deployment on Astronomer."
...
-* Create or update an existing Airflow Deployment on Astronomer with that appended IAM role
-* Confirm your deployment was annotated successfully by running `kubectl describe po` on any new pod created within that namespace
+IAM roles on [AWS](https://aws.amazon.com/iam/faqs/) and other platforms are often used to manage the level of access a specific user (or object, or group of users) have to some resource (or set of resources). The resource in question could be an S3 bucket or Secret Backend, both of which are commonly used in tandem with Airflow and Astronomer and can now be configured to be accessible only to a subset of Kubernetes pods within your wider Astronomer cluster.
 
 A few clarifying notes:
 
-* ALL Kubernetes pods within your Astronomer Deployment (Scheduler, Webserver, Workers) will assume the IAM role. There is currently no way to run more than 1 IAM role per deployment.
+* Webserver, Scheduler and Worker pods within your Airflow Deployment will assume the IAM role. There is currently no way to use more than 1 IAM role per deployment.
```
Comment (paolaperaza) [inline enterprise/v0.16/05_customize-airflow/06_integrate-iam.md:15]: *per `Airflow Deployment` (know it was deployment before, think we're just tryna standardize how we use that term)

File: enterprise/v0.16/05_customize-airflow/06_integrate-iam.md [hunk 3]
```diff
@@ -6,151 +6,234 @@ description: "How to append IAM roles to an Airflow Deployment on Astronomer."
...
+$ astro deployment update <deployment-name> --cloud-role=arn:aws:iam::123456789:role/example-role
 ```
 
-As you run these commands make sure to:
-
-- Insert your own `role-arn` for the IAM role in question after the `--cloud-role` flag
-- Specify your Airflow Executor (Celery, Local, Kubernetes) if you create a Deployment
-- Specify the release name you'd like to give to your new Airflow Deployment if applicable (e.g. `new-deployment-1234)
-
-> **Note:** To customize an Airflow Deployment's release name when you create it, make sure you've set `manualReleaseNames: true` in the `config.yaml` file of your Astronomer directory. Enabling that configuration should make applying the IAM role to that Airflow Deployment easier to automate. If you _don't_ enable `manualReleaseNames`, the release name for any Aiflow Deployment you create will be generated automatically and at random.
```
Comment (paolaperaza) [inline enterprise/v0.16/05_customize-airflow/06_integrate-iam.md:140]: Worth bringing back this manual release name part or nah?

**Review by paolaperaza — APPROVED** [review]
This looks great, @ianstanton ! Awesome improvements across the board - truly. I'll defer to @vparekh94 before merging this but I really didn't have much feedback formatting or copy wise.

Great work, thanks for taking initiative 😊

File: enterprise/v0.16/05_customize-airflow/06_integrate-iam.md (hunk 3 above)
Comment (ianstanton) [reply enterprise/v0.16/05_customize-airflow/06_integrate-iam.md:140]: I didn't feel this piece was relevant here. I vote to keep it out.

Comment (paolaperaza) [reply enterprise/v0.16/05_customize-airflow/06_integrate-iam.md:140]: That works, no problem with me

File: enterprise/v0.16/05_customize-airflow/06_integrate-iam.md (hunk 2 above)
Comment (ianstanton) [reply enterprise/v0.16/05_customize-airflow/06_integrate-iam.md:15]: Did you want it to read `Astronomer Deployment`?

Comment (paolaperaza) [reply enterprise/v0.16/05_customize-airflow/06_integrate-iam.md:15]: `Airflow Deployment`! Was it that way originally? Might have linked to the wrong line. Either way - not a huge deal 🙂

File: enterprise/v0.15/05_customize-airflow/06_integrate-iam.md [hunk 4]
```diff
@@ -6,68 +6,234 @@ description: "How to append IAM roles to an Airflow Deployment on Astronomer."
...
-
-To support this functionality, each pod in an Astronomer namespace (each of which maps to an Airflow deployment) is launched by a [Kubernetes Service Account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/). By annotating the service account with the role, all pods launched will inherit that role.
-
-At a high-level, you can now:
-
-* Create an IAM role defining access to the target service or set of services (e.g. AWS S3)
-* Create or update an existing Airflow Deployment on Astronomer with that appended IAM role
-* Confirm your deployment was annotated successfully by running `kubectl describe po` on any new pod created within that namespace
+IAM roles on [AWS](https://aws.amazon.com/iam/faqs/) and other platforms are often used to manage the level of access a specific user (or object, or group of users) have to some resource (or set of resources). The resource in question could be an S3 bucket or Secret Backend, both of which are commonly used in tandem with Airflow and Astronomer and can now be configured to be accessible only to a subset of Kubernetes pods within your wider Astronomer cluster.
```
Comment (ben-astro) [inline enterprise/v0.15/05_customize-airflow/06_integrate-iam.md:11]: grammatical issue - "level of access a specific user (...) _has_ to some resource...."

**Review by sunkickr — COMMENTED** [review]
All the changes look good to me!

**Review by sunkickr — APPROVED** [review]

File: enterprise/v0.15/05_customize-airflow/06_integrate-iam.md [hunk 5]
```diff
@@ -6,68 +6,234 @@ description: "How to append IAM roles to an Airflow Deployment on Astronomer."
...
+  houston:
+    config:
+      deployments:
+        serviceAccountAnnotationKey: eks.amazonaws.com/role-arn
 ```
 
-Make sure to insert your own `role-arn` for the IAM role in question after the `--cloud-role` flag.
+2. Run `helm upgrade` to apply changes.
+```bash
+$ helm upgrade astronomer -f config.yaml --version=0.15.8 astronomer/astronomer --namespace astronomer
```
Comment (ben-astro) [inline enterprise/v0.15/05_customize-airflow/06_integrate-iam.md:205]: Should we update the version in here to be latest? 0.16.X ?

Comment (ianstanton) [reply enterprise/v0.15/05_customize-airflow/06_integrate-iam.md:205]: This line is for the `v0.15` docs page. There are two files here (one for `v0.15.x` and one for `v0.16.x`).

File: enterprise/v0.15/05_customize-airflow/06_integrate-iam.md (hunk 4 above)
Comment (ianstanton) [reply enterprise/v0.15/05_customize-airflow/06_integrate-iam.md]: Thanks! Fixed in both versions.

**Review by vparekh94 — APPROVED** [review]

**Pull request merged**